spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/util/ConfigCheck.scala

```diff
@@ -29,7 +32,11 @@ object ConfigCheck {
    * @param conf SparkConf object to check
    */
   def checkConfig(conf: SparkConf): Unit = {
-    val unknownProps = getUnknownProperties(conf)
+    val connectionFactory = CassandraConnectionFactory.fromSparkConf(conf)
```

Comment:
Ok, we can address it in another ticket. On the other hand, multiple apply() overload with *default* parameters is not allowed, while naming them differently allows default params.